### DISCUSSION TITLE
[4.x]: URL field should allow for in-page anchor links

### DISCUSSION THREAD

**Discussion by thisisjamessmith** [craftcms/cms#12051] [Ideas]
### What happened?

In my opinion the URL fieldtype should allow for in-page anchor links. I can't think of a good reason why not, at any rate:

<img width="194" alt="image" src="https://user-images.githubusercontent.com/2416106/192546598-bcb79994-f955-44ee-ada6-9fc2ef147781.png">

### Craft CMS version

4.2.5.1

### Installed plugins and versions

-

**Comment by brianjhanson** [comment]
Thank you! This is currently working as expected so I've converted it to a discussion where others can weigh in and we can evaluate if we should make this change.

**Comment by jannisborgers** [comment]
Late to the party. It’s the first time I use URL fields after the Typed Link Field plugin has been abandoned. 

I would appreciate making anchors possible in URL fields. It is the easiest way to let authors add in-page-navigation. Right now, because the URL get’s run through `isFullUrl()`, it gets prepended with `http://`. As a workaround, this is what I do:

```twig
{{ entry.link_url starts with "http://#" ? entry.link_url|replace("http://#", "#") : entry.link_url }}
```

🤷‍♂️
